File: ffi/capi/bindings/js/CollatorStrength.mjs
import wasm from "./diplomat-wasm.mjs"
import * as diplomatRuntime from "./diplomat-runtime.mjs"

export const CollatorStrength_js_to_rust = {
  "Auto": 0,
  "Primary": 1,
  "Secondary": 2,
  "Tertiary": 3,
  "Quaternary": 4,
  "Identical": 5,
};

export const CollatorStrength_rust_to_js = {
  [0]: "Auto",
  [1]: "Primary",
  [2]: "Secondary",
  [3]: "Tertiary",
  [4]: "Quaternary",
  [5]: "Identical",
};

export const CollatorStrength = {
  "Auto": "Auto",
  "Primary": "Primary",
  "Secondary": "Secondary",
  "Tertiary": "Tertiary",
  "Quaternary": "Quaternary",
  "Identical": "Identical",
};

File: ffi/mvn/src/main/kotlin/src/main/kotlin/org/unicode/icu4x/CollatorStrength.kt
package org.unicode.icu4x

import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface CollatorStrengthLib: Library {
}
/** See the [Rust documentation for `Strength`](https://docs.rs/icu/2.1.1/icu/collator/options/enum.Strength.html) for more information.
*/
enum class CollatorStrength {
    Primary,
    Secondary,
    Tertiary,
    Quaternary,
    Identical;

    fun toNative(): Int {
        return this.ordinal
    }


    companion object {
        internal val libClass: Class<CollatorStrengthLib> = CollatorStrengthLib::class.java
        internal val lib: CollatorStrengthLib = Native.load("icu4x", libClass) 
        fun fromNative(native: Int): CollatorStrength {
            return CollatorStrength.entries[native]
        }

        fun default(): CollatorStrength {
            return Primary
        }
    }
}
